File: LeetCode/070.climbing-stairs/climbing-stairs.java
class Solution {
    private int FibHelper(int n, int[] array){
        if (n < 3)  return n;
        else if (array[n] == 0){
            array[n] = FibHelper(n - 1, array) + FibHelper(n - 2, array);
        }
        return array[n];
    }

    public int climbStairs(int n) {
        int[] array = new int[n + 1];
        return FibHelper(n, array);
    }
}

File: LeetCode/0070.climbing-stairs/climbing-stairs.cpp
class Solution {
private:
    int FibHelper(int n, vector<int>&res){
        if (n < 3)  return n;
        else if (res[n] == 0){
            res[n] = FibHelper(n - 1, res) + FibHelper(n - 2, res);
        }
        return res[n];
    }
public:
    int climbStairs(int n){
        vector<int>res;
        res.resize(n + 1);
        return FibHelper(n, res);
    }
};
